src/Illuminate/Database/Eloquent/Relations/Pivot.php

```diff
@@ -92,6 +105,10 @@ public static function fromRawAttributes(Model $parent, $attributes, $table, $ex
      */
     protected function setKeysForSaveQuery(Builder $query)
     {
+        if ($this->instantiatedAsModel) {
```

Comment:
hmm, to me, this screams the need for class extension, to avoid runtime mimicry of what should really be a superclass